packages/blade/src/components/FileUpload/__tests__/FileUpload.test.stories.tsx

```diff
@@ -0,0 +1,449 @@
+  ]);
+  const file1 = new File([blob], 'my-image.png', {
+    type: 'image/png',
+  });
+  const file2 = new File([blob], 'my-image2.png', {
+    type: 'image/png',
+  });
+  const file3 = new File([blob], 'my-image3.png', {
+    type: 'image/png',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
+
+  await expect(input).toHaveAttribute('multiple');
+
+  await userEvent.upload(input as HTMLElement, [file1, file2, file3]);
+  await expect(getByText("You can't upload more than 2 files.")).toBeInTheDocument();
+  await expect(queryByText('my-image.png')).not.toBeInTheDocument();
+  await expect(queryByText('my-image2.png')).not.toBeInTheDocument();
+};
+
+export const TestMaxSizeFileUpload: StoryFn<typeof Toast> = (): React.ReactElement => {
+  return (
+    <Box maxWidth="400px">
+      <FileUpload
+        uploadType="single"
+        label="Upload GST certificate"
+        helpText=" .jpg, .jpeg, or .png file only"
+        accept="image/*"
+        isRequired
+        // 10 bytes for testing purpose
+        maxSize={10}
+        necessityIndicator="required"
+      />
+    </Box>
+  );
+};
+
+TestMaxSizeFileUpload.play = async () => {
+  const { getByText } = within(document.body);
+
+  // Create a file
+  const blob = new Blob([
+    'data:image/png;base64,R0lGODlhDwAPAKECAAAAzMzM/////wAAACwAAAAADwAPAAACIISPeQHsrZ5ModrLlN48CXF8m2iQ3YmmKqVlRtW4MLwWACH+H09wdGltaXplZCBieSBVbGVhZCBTbWFydFNhdmVyIQAAOw==',
+  ]);
+  const filename = 'my-image.png';
+  const file = new File([blob], filename, {
+    type: 'image/png',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
+
+  await expect(input).not.toHaveAttribute('multiple');
+
+  await userEvent.upload(input as HTMLElement, file);
+  await expect(getByText('File size exceeded.')).toBeInTheDocument();
+};
+
+export const TestAcceptProp: StoryFn<typeof Toast> = (): React.ReactElement => {
+  return (
+    <Box maxWidth="400px">
+      <FileUpload
+        uploadType="single"
+        label="Upload GST certificate"
+        helpText=" .jpg, .jpeg, or .png file only"
+        accept="image/*"
+        isRequired
+        necessityIndicator="required"
+      />
+    </Box>
+  );
+};
+
+TestAcceptProp.play = async () => {
+  const { getByText, queryByText } = within(document.body);
+
+  // Create a file
+  const blob = new Blob([
+    'data:application/pdf;base64,R0lGODlhDwAPAKECAAAAzMzM/////wAAACwAAAAADwAPAAACIISPeQHsrZ5ModrLlN48CXF8m2iQ3YmmKqVlRtW4MLwWACH+H09wdGltaXplZCBieSBVbGVhZCBTbWFydFNhdmVyIQAAOw==',
+  ]);
+  const filename = 'my-pdf.pdf';
+  const file = new File([blob], filename, {
+    type: 'application/pdf',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
+
+  await expect(input).not.toHaveAttribute('multiple');
+
+  await userEvent.upload(input as HTMLElement, file);
+  await expect(queryByText(filename)).not.toBeInTheDocument();
```

Comment:
assert on error maybe like you did it in the test above this?

Reply:
In this case, the error is not visible. It's visible only in case of drag & drop. In case of select, the pdf files are just not selectable in the dialouge.